src/Mvc/Mvc.Razor/test/RazorPageCreateModelExpressionTest.cs

```diff
@@ -100,22 +61,53 @@ public static TheoryData NotQuiteIdentityExpressions
             var page = CreateIdentityPage(viewContext);
 
             // Act
-            var modelExpression = createModelExpression(page);
+            var modelExpression = page.CreateModelExpression2();
 
             // Assert
             Assert.NotNull(modelExpression);
-            Assert.Equal(expectedName, modelExpression.Name);
+            Assert.Empty(modelExpression.Name);
             Assert.Same(modelExplorer, modelExpression.ModelExplorer);
         }
 
-        [Theory]
-        [MemberData(nameof(NotQuiteIdentityExpressions))]
-        public void CreateModelExpression_ReturnsExpectedMetadata_NotQuiteIdentityExpressions(
+        [Fact]
+        public void CreateModelExpression_ReturnsExpectedMetadata_NotQuiteIdentityExpressions_ForModelGivesMDotModel()
+        {
+            // m => m.Model
+            // Arrange
+            var expectedName = "Model";
+            var expectedType = typeof(RecursiveModel);
+
+            CreateModelExpression_NotQuiteIdentityExpressions(page => page.CreateModelExpression1(), expectedName, expectedType);
+        }
+
+        [Fact]
+        public void CreateModelExpression_ReturnsExpectedMetadata_NotQuiteIdentityExpressions_ForModelGivesViewDataDotModel()
+        {
+            // m => ViewData.Model
+            // Arrange
+            var expectedName = "ViewData.Model";
+            var expectedType = typeof(RecursiveModel);
+
+            CreateModelExpression_NotQuiteIdentityExpressions(page => page.CreateModelExpression2(), expectedName, expectedType);
+        }
+
+        [Fact]
+        public void CreateModelExpression_ReturnsExpectedMetadata_NotQuiteIdentityExpressions_ForModelGivesViewContextDotViewDataDotModel()
+        {
+            // m => ViewContext.ViewData.Model
+            // Arrange
+            var expectedName = "ViewContext.ViewData.Model";
+            // This property has type object because ViewData is not exposed as ViewDataDictionary<TModel>.
+            var expectedType = typeof(object);
+
+            CreateModelExpression_NotQuiteIdentityExpressions(page => page.CreateModelExpression3(), expectedName, expectedType);
+        }
+
+        private static void CreateModelExpression_NotQuiteIdentityExpressions(
```

Comment:
Like this approach because it makes it very clear what changes between the three tests. Not blocking but why didn't you use something similar for the `int` and `string`-related tests?

Reply:
Just being lazy. It was easier to copy paste :)